plugin/src/main/java/io/jenkins/plugins/analysis/core/steps/IssuesPublisher.java

```diff
@@ -243,4 +277,64 @@ private QualityGateEvaluationMode determineQualityGateEvaluationMode(final Repor
         }
         return qualityGateEvaluationMode;
     }
+
+    private String extractChecksSummary(final IssuesStatistics statistics) {
+        return String.format("## %d issues in total:\n"
+                        + "- ### %d new issues\n"
+                        + "- ### %d Outstanding Issues\n"
+                        + "- ### %d delta issues\n"
+                        + "- ### %d fixed issues",
+                statistics.getTotalSize(), statistics.getNewSize(), statistics.getTotalSize() - statistics.getNewSize(),
+                statistics.getDeltaSize(), statistics.getFixedSize());
+    }
+
+    private String extractChecksText(final IssuesStatistics statistics) {
+        return "## Total Issue Statistics:\n"
+                + generateSeverityText(statistics.getTotalLowSize(), statistics.getTotalNormalSize(),
+                statistics.getTotalHighSize(), statistics.getTotalErrorSize())
+                + "## New Issue Statistics:\n"
+                + generateSeverityText(statistics.getNewLowSize(), statistics.getNewNormalSize(),
+                statistics.getNewHighSize(), statistics.getNewErrorSize())
+                + "## Delta Issue Statistics:\n"
+                + generateSeverityText(statistics.getDeltaLowSize(), statistics.getDeltaNormalSize(),
+                statistics.getDeltaHighSize(), statistics.getDeltaErrorSize());
+    }
+
+    private String generateSeverityText(final int low, final int normal, final int high, final int error) {
+        return "* low: " + low + "\n"
+                + "* Normal: " + normal + "\n"
+                + "* High: " + high + "\n"
+                + "* Error: " + error + "\n";
+    }
+
+    private ChecksConclusion extractChecksConclusion(final QualityGateStatus status) {
+        switch (status) {
+            case INACTIVE:
+                return ChecksConclusion.SUCCESS;
+            case FAILED:
+                return ChecksConclusion.FAILURE;
+            case WARNING:
+                return ChecksConclusion.NEUTRAL;
```

Comment:
I don't think Neutral is the same as Warning

possibly:
```suggestion
                return ChecksConclusion.FAILURE;
```

Reply:
WARNING vs. FAILED is a Jenkins specific concept that does not exist in Github